ivy_tests/test_ivy/test_functional/test_core/test_linalg.py

```diff
@@ -1473,6 +1473,50 @@ def test_cholesky(
     )
 
 
+# cov
+@handle_test(
+    fn_tree="functional.ivy.cov",
+    dtype_x1_x2=dtype_value1_value2_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=2,
+        max_dim_size=5,
+        min_value=-1e10,
+        max_value=1e10,
+        abs_smallest_val=0.01,
+        large_abs_safety_factor=2,
+    ),
+    num_positional_args=helpers.num_positional_args(fn_name="cov"),
+)
+def test_cov(
+    *,
+    dtype_x1_x2,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    fn_name,
+    container_flags,
+    instance_method,
+    fw,
+):
+    dtype, x1, x2, _ = dtype_x1_x2
+    helpers.test_function(
+        input_dtypes=dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        container_flags=container_flags,
+        instance_method=instance_method,
+        fw=fw,
+        fn_name=fn_name,
+        x1=x1,
```

Comment:
please add:
`on_device=on_device`
`test_gradients=True`

between the `fn_name` and `x1` arguments